### ISSUE TITLE
Add ability to query current peer reputation

### ISSUE THREAD

**Issue by nazar-pc** [paritytech/polkadot-sdk#2185]
### Is there an existing issue?

- [X] I have searched the existing issues

### Experiencing problems? Have you tried our Stack Exchange first?

- [X] This is not a support question.

### Motivation

In a custom gossip protocol we'd like to make some decisions based on peer reputation. We already increase/decrease reputation according to peer's communication, but can't query current reputation of the peer using public methods without re-implementing `sc_service::build_network`, which is an annoying amount of code.

### Request

Introduce `NetworkPeers::peer_reputation(peer_id)` or similar API such that it is possible to query current reputation of a peer by their `PeerId`.

### Solution

Awkward thing is that right now `NetworkPeers` primarily uses `to_worker` channel to send one-way notifications to network worker, so I'm not entirely sure what would be a clean solution to access peer reputation here.

### Are you willing to help with this request?

Yes!

**Comment by bkchr** [comment]
CC @paritytech/networking

**Comment by altonen** [comment]
It should be pretty easy to do. You can store `PeerStoreHandle` to `NetworkService` and then implement `NetworkPeers::peer_reputatation(peer)` which would just call the `PeerStoreHandle`.

**Comment by nazar-pc** [comment]
Yes, that was my first idea too, but then why other operations on `PeerStoreHandle` like `.report_peer()` call go through a channel with `ServiceToWorkerMsg` messages? Should those be refactored as well to call `PeerStoreHandle` directly?

**Comment by nazar-pc** [comment]
PR: https://github.com/paritytech/polkadot-sdk/pull/2392

**Comment by altonen** [comment]
Since writing my last comment, I did some load testing for unrelated stuff and those results have brought into question our whole reputation system (#2257). We're going to re-evaluate it in the near future because it's not giving us a lot of benefit for the cost it takes to keep it running. Unless this is a really critical feature for you that you're unable to keep in your fork, I'd advice against using this system until we've figured out in our end what to do with it.
